### ISSUE TITLE
Debugging

### ISSUE THREAD

**Issue by Sampurn44** [microsoft/vscode-cpptools#9677]
### Environment

- OS and version: 22000.795 & 21H2
- VS Code:1.69.2
- C/C++ extension: Name: C/C++
Id: ms-vscode.cpptools
Description: C/C++ IntelliSense, debugging, and code browsing.
Version: 1.12.0
Publisher: Microsoft
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools
- OS and version of remote machine (if applicable):
- GDB / LLDB version:7.6.1

### Bug Summary and Steps to Reproduce

Bug Summary: Everytime I Debug a code on vscode the terminal takes a pause for somes sec and terminal gives the output 'c:\Users\sampu.vscode\extensions\ms-vscode.cpptools-1.11.4-win32-x64\debugAdapters\bin\WindowsDebugLauncher.exe' '--stdin=Microsoft-MIEngine-In-5or3txdx.55l' '--stdout=Microsoft-MIEngine-Out-2mbbrlkj.0my' '--stderr=Microsoft-MIEngine-Error-mn22flnn.vrz' '--pid=Microsoft-MIEngine-Pid-bdxzj0g3.l1b' '--dbgExe=C:\MinGW\bin\gdb.exe' '--interpreter=mi'

### Debugger Configurations

```shell
{
    "tasks": [
        {
            "type": "cppbuild",
            "label": "C/C++: g++.exe build active file",
            "command": "C:\\MinGW\\bin\\g++.exe",
            "args": [
                "-fdiagnostics-color=always",
                "-g",
                "${file}",
                "-o",
                "${fileDirname}\\${fileBasenameNoExtension}.exe"
            ],
            "options": {
                "cwd": "C:\\MinGW\\bin"
            },
            "problemMatcher": [
                "$gcc"
            ],
            "group": {
                "kind": "build",
                "isDefault": true
            },
            "detail": "Task generated by Debugger."
        },
        {
            "type": "cppbuild",
            "label": "C/C++: cpp.exe build active file",
            "command": "C:\\MinGW\\bin\\cpp.exe",
            "args": [
                "-fdiagnostics-color=always",
                "-g",
                "${file}",
                "-o",
                "${fileDirname}\\${fileBasenameNoExtension}.exe"
            ],
            "options": {
                "cwd": "${fileDirname}"
            },
            "problemMatcher": [
                "$gcc"
            ],
            "group": "build",
            "detail": "Task generated by Debugger."
        }
    ],
    "version": "2.0.0"
}
```

### Debugger Logs

```shell
Debug Console shows this
Loaded 'C:\WINDOWS\SysWOW64\kernel32.dll'. Symbols loaded.
Loaded 'C:\WINDOWS\SysWOW64\KernelBase.dll'. Symbols loaded.
Loaded 'C:\WINDOWS\SysWOW64\msvcrt.dll'. Symbols loaded.
Loaded 'C:\MinGW\bin\libgcc_s_dw2-1.dll'. Symbols loaded.
Loaded 'C:\MinGW\bin\libstdc++-6.dll'. Symbols loaded.
[New Thread 19372.0xf00]
1023
COLD
The program 'c:\Users\sampu\Desktop\vscode\hello1.exe' has exited with code 0 (0x00000000).
and the terminal shows
c:\Users\sampu.vscode\extensions\ms-vscode.cpptools-1.11.4-win32-x64\debugAdapters\bin\WindowsDebugLauncher.exe' '--stdin=Microsoft-MIEngine-In-5or3txdx.55l' '--stdout=Microsoft-MIEngine-Out-2mbbrlkj.0my' '--stderr=Microsoft-MIEngine-Error-mn22flnn.vrz' '--pid=Microsoft-MIEngine-Pid-bdxzj0g3.l1b' '--dbgExe=C:\MinGW\bin\gdb.exe' '--interpreter=mi'
```

### Other Extensions

<html>
<body>

Extension | Author (truncated) | Version
-- | -- | --
doxdocgen | csc | 1.4.0
code-runner | for | 0.11.8
better-cpp-syntax | jef | 1.15.19
cmake-tools | ms- | 1.11.26
cpptools | ms- | 1.11.4
cpptools-extension-pack | ms- | 1.2.0
cmake | twx | 0.0.17

<p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; color: rgb(201, 209, 217); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">(1 theme extensions excluded)</p>
</body>
</html>

### Additional Information

https://user-images.githubusercontent.com/92144348/182447164-920f1076-6840-48ed-8d37-ff90c7b5fcdf.mp4

**Comment by WardenGnaw** [comment]
The command: 
> 'c:\Users\sampu.vscode\extensions\ms-vscode.cpptools-1.11.4-win32-x64\debugAdapters\bin\WindowsDebugLauncher.exe' '--stdin=Microsoft-MIEngine-In-5or3txdx.55l' '--stdout=Microsoft-MIEngine-Out-2mbbrlkj.0my' '--stderr=Microsoft-MIEngine-Error-mn22flnn.vrz' '--pid=Microsoft-MIEngine-Pid-bdxzj0g3.l1b' '--dbgExe=C:\MinGW\bin\gdb.exe' '--interpreter=mi'

That is how the extension is able to use the integrated terminal of VS Code.

**Comment by Sampurn44** [comment]
@WardenGnaw 
But I am not able to code as shown in the video can u help?

**Comment by WardenGnaw** [comment]
The program is waiting on input based on your code `cin >> w` which you can type in the terminal pane if you are using the integrated terminal.

**Comment by Sampurn44** [comment]
@WardenGnaw i can't type any input in the terminal

**Comment by Gurpreet-Singh121** [comment]
facing exactly same error please help me sir

**Comment by Goklk** [comment]
Bro did you find a way to solve this??

**Comment by AdityaRajaram13** [comment]
getting same error
any solution?

**Comment by Goklk** [comment]
No bro, i have the same problem.

**Comment by Goklk** [comment]
Bro I installed powershell new version and installed powershell for vs code extension. Now it's working

**Comment by diyaaar** [comment]
i have the same problem, did anyone solved it?

**Comment by Goklk** [comment]
Yes for me it's working now

**Comment by diyaaar** [comment]
may you share how did you do it?

**Comment by Goklk** [comment]
Installed powershell in vs code

**Comment by Goklk** [comment]
Also see the variable environment and add your path to compiler

**Comment by Dru01** [comment]
@Goklk can you give more detail, please. I've allready update powershell to 7th and install powershell extension to vs-code. The main problem is that when I try to debug a c++ program when i get to the point to step over a cin statement it freezes if i've allready enter the input. If i wait until a get there and then enter the input there is no problem...

**Comment by SachinShantharam** [comment]
Me too facing the same issue. Any solution ?

**Comment by harshdobariya** [comment]
Same issue, 


- Debug line not highlighting

- No variables showing in local/global variables. 

-  Not stopping at breakpoints as expected.

launch.json
`{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",

    "configurations": [
        {
            "name": "(gdb) Launch",
            "type": "cppdbg",
            "request": "launch",
            "program": "${fileDirname}/${fileBasenameNoExtension}",
            "args": [],
            "stopAtEntry": true,
            "cwd": "${fileDirname}",
            "environment": [],
            "externalConsole": false,
            "MIMode": "gdb",
            "miDebuggerPath": "",
            "setupCommands": [
                {
                    "description": "Enable pretty-printing for gdb",
                    "text": "-enable-pretty-printing",
                    "ignoreFailures": true
                },
                {
                    "description": "Set Disassembly Flavor to Intel",
                    "text": "-gdb-set disassembly-flavor intel",
                    "ignoreFailures": true
                }
            ]
        },

    ]
}`

cpp Code

`#include<iostream>
#include<string>

using namespace std;

int main() {
    string s;

    cout << "Hello world.\n";
    int g = 85;
    cin >> s;
    cout << s << " 55" << endl;

    return 0;
}

`

Debug Concole
`Thread 1 hit Breakpoint 1, 0x00007ff773bf1460 in main ()
Loaded 'C:\WINDOWS\SYSTEM32\ntdll.dll'. Symbols loaded.
Loaded 'C:\WINDOWS\System32\kernel32.dll'. Symbols loaded.
Loaded 'C:\WINDOWS\System32\KernelBase.dll'. Symbols loaded.
Loaded 'C:\WINDOWS\System32\msvcrt.dll'. Symbols loaded.
Loaded 'C:\msys64\mingw64\bin\libgcc_s_seh-1.dll'. Symbols loaded.
Loaded 'C:\msys64\mingw64\bin\libstdc++-6.dll'. Symbols loaded.
Loaded 'C:\msys64\mingw64\bin\libwinpthread-1.dll'. Symbols loaded.
Loaded 'C:\WINDOWS\SYSTEM32\cryptbase.dll'. Symbols loaded.
Loaded 'C:\WINDOWS\System32\bcryptprimitives.dll'. Symbols loaded.
Execute debugger commands using "-exec <command>", for example "-exec info registers" will list registers in use (when GDB is the debugger)
Single stepping until exit from function main,
which has no line number information.
[Thread 6360.0x4154 exited with code 0]
[Thread 6360.0x6d4 exited with code 0]
[Thread 6360.0x30c8 exited with code 0]`


![cpp debug bug](https://github.com/microsoft/vscode-cpptools/assets/18502227/2cd56b77-9755-4ee9-adb5-fb9eaf164417)

**Comment by Kundan525** [comment]
I am also getting the same error. Could anybody help me!

**Comment by Vuong-Chu** [comment]
Use **tasks.json**

{
  "version": "2.0.0",
  "tasks": [
    {
      "type": "shell",
      "label": "C/C++: cl.exe build active file",
      "command": "cl.exe",
      "args": [
        "/Zi",
        "/EHsc",
        "/Fe:",
        "${fileDirname}\\${fileBasenameNoExtension}.exe",
        "${file}"
      ],
      "problemMatcher": ["$msCompile"],
      "group": {
        "kind": "build",
        "isDefault": true
      },
      "detail": "Task generated by Debugger."
    }
  ]
}

Use **launch.json**

{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "C/C++: cl.exe build and debug active file",
      "type": "cppvsdbg",
      "request": "launch",
      "program": "${fileDirname}\\${fileBasenameNoExtension}.exe",
      "args": [],
      "stopAtEntry": false,
      "cwd": "${workspaceFolder}",
      "environment": [],
      "externalConsole": false,
      "preLaunchTask": "C/C++: cl.exe build active file"
    }
  ]
}

**Use c_cpp_properties.json**
{
  "configurations": [
    {
      "name": "Win32",
      "includePath": ["${workspaceFolder}/**"],
      "defines": ["_DEBUG", "UNICODE", "_UNICODE"],
      "windowsSdkVersion": "10.0.18362.0",
      "compilerPath": "C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.24.28314/bin/Hostx64/x64/cl.exe",
      "cStandard": "c11",
      "cppStandard": "c++17",
      "intelliSenseMode": "msvc-x64"
    }
  ],
  "version": 4
}

**Comment by Aura4G** [comment]
I think I know what triggers the error, having gotten this issue as well. I believe it is based on the type of compiler you first build the program with when debugging the script. The original post contained MinGW in the script? I used that tool to install the gcc compiler for c++. I then debugged my program with g++ and got that error. I could not fix it for that script in particular, but when I opened a new script and debugged it with the cl compiler, it worked.

So here is what I have determined:
- Compiling the program with g++ or cpp .exes do not require you to open vs code through the dev cmd, but you cannot run the program with these compilers in the vs code terminal
   - I noticed that using these compilers will not create the extra files that are typically made with cl (pdb and ilk files)
- Compiling with cl works in the vs code terminal, but requires the dev command prompt to be used first.

I can't yet determine which is more efficient to use
